frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,166 @@
+  const { close, cancel, resource, handlePromise, errorMessage, inProgress } = props;
+  const pvcName: string = resource.metadata.name;
+  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+
+  const pvcUsedCapacityQuery: string = getPVCUsedCapacityQuery(pvcName);
+
+  const [response, error, loading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    query: pvcUsedCapacityQuery,
+    namespace: resource.metadata.namespace,
+  });
+
+  const pvcUsedCapacityQueryResult: DataPoint[] = getInstantVectorStats(
+    response,
+    null,
+    humanizeBinaryBytes,
+  );
+
+  const pvcUsedCapacity = pvcUsedCapacityQueryResult?.[0]?.label || 'No Data';
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+
+    const pvcCloneObj = {
+      apiVersion: PersistentVolumeClaimModel.apiVersion,
+      kind: PersistentVolumeClaimModel.kind,
+      metadata: {
+        name: clonePVCName,
+        namespace: resource.metadata.namespace,
+      },
+      spec: {
+        storageClassName: resource.spec.storageClassName,
+        dataSource: {
+          name: pvcName,
+          kind: PersistentVolumeClaimModel.kind,
+          apiGroup: '',
+        },
+        accessModes: resource.spec.accessModes,
+        resources: resource.spec.resources,
+      },
+    };
+
+    return handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj)).then(close);
+  };
+
+  return (
+    <Form onSubmit={submit}>
+      <div className="modal-content modal-content--no-inner-scroll">
+        <ModalTitle>Clone</ModalTitle>
+        <ModalBody>
+          <FormGroup label="Name" isRequired fieldId="ceph-clone-pvc-modal__name">
+            <TextInput
+              isRequired
+              type="text"
+              id="ceph-clone-pvc-modal__name"
+              name="ceph-clone-pvc-modal__name"
+              value={clonePVCName}
+              onChange={setClonePVCName}
+            />
+          </FormGroup>
+          <div className="ceph-clone-pvc-modal__details">
+            <label htmlFor="ceph-clone-pvc-modal__details-section" className="text-muted">
+              PVC Details
+            </label>
+            <div
+              id="ceph-clone-pvc-modal__details-section"
+              className="ceph-clone-pvc-modal__details-section"
+            >
+              <div>
+                <div>
+                  <label htmlFor="ceph-clone-pvc-modal__namespace">Namespace</label>
+                  <p id="ceph-clone-pvc-modal__namespace">
+                    <ResourceIcon kind="Namespace" />
+                    {resource.metadata.namespace}
+                  </p>
+                </div>
+                <div>
+                  <label htmlFor="ceph-clone-pvc-modal__storageclass">Storage Class</label>
+                  <p id="ceph-clone-pvc-modal__storageclass">
+                    <ResourceIcon kind="StorageClass" />
+                    {resource.spec.storageClassName || 'NA'}
```

Comment:
"NA"? Is that for "not applicable"? "None" is probably better. We shouldn't show the icon if there is no storage class.